### PR TITLE
Added refill rate parameter so we can control how long instrument remains throttled for

### PR DESCRIPTION
**Pull request by clare**

### CODE REVIEW

File: _release/thermal-recorder.yaml [hunk 1]
```diff
@@ -88,3 +88,6 @@ throttler:
     # if value is less than the min recording length then it will set to min recording length
     sparse-length-secs: 30
 
+    # rate that throttling is removed.  Compared with '1.0', a value of '2.0' mean that throttling is applied for half
+    # the time, while a value '0.5' will mean throttling is applied for twice as long
```
Comment (mjs) [inline _release/thermal-recorder.yaml]: Suggested rewording:
```
The controls how soon normal recording resumes once throttling is active. Compared with '1.0', a value of '2.0' means that normal recording resumes in half the time, while a value '0.5' will means throttling remains active for double the time.
```

File: throttle/tokenbucket_test.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ type TestBucket struct {
 	TokenBucket
 }
 
-func (b *TestBucket) hasExactly(tokens uint32) bool {
+func (b *TestBucket) hasExactly(tokens float64) bool {
```
Comment (mjs) [inline throttle/tokenbucket_test.go]: Now that tokens are being tracked as floats (which I agree was the right thing to do), exact values are less likely, especially with fractional refill rates. Where and how is hasExactly used?

**Review by mjs — COMMENTED** [review]
Looking good. I'm a little concerned about hasExactly.

File: _release/thermal-recorder.yaml (hunk 1 above)
Comment (clare) [reply _release/thermal-recorder.yaml]: Thanks, I was struggling to explain this.

File: throttle/tokenbucket_test.go (hunk 2 above)
Comment (clare) [reply throttle/tokenbucket_test.go]: I've renamed hasExactly to hasApprox.  It is used to test the value in the bucket is approximately what is expected.

**Review by mjs — APPROVED** [review]

**Pull request merged**